### PR TITLE
Add option to copy sender/recipient email addresses to clipboard

### PR DESCRIPTION
**Pull request by Riverside14**
Hi,
I add an option to copy email addresses from the message header to clipboard.

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java [hunk 1]
```diff
@@ -130,6 +136,8 @@ protected void onFinishInflate() {
 
     @Override
     public void onClick(View view) {
+        Toast toast = Toast.makeText(view.getContext(), "bonjour", Toast.LENGTH_SHORT);
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java]: Left over from debugging?

**Comment by Riverside14** [comment]
Yes it was, i just removed it. Sorry.
I hope everything is correct.

File: k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageTopView.java [hunk 2]
```diff
@@ -126,6 +126,10 @@ public void setOnToggleFlagClickListener(OnClickListener listener) {
         mHeaderContainer.setOnFlagListener(listener);
     }
 
+    public void setOnToggleFlagLongClickListener(OnLongClickListener listener){
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageTopView.java]: This change seems unrelated to copying email addresses to the clipboard. Please limit the scope of pull requests as much as possible.

File: k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java [hunk 3]
```diff
@@ -143,6 +149,31 @@ public void onClick(View view) {
...
     }
 
+    @Override
+    public boolean onLongClick(View view) {
+        switch (view.getId()) {
+            case R.id.from:
+                onAddAddressesToClipboard(mMessage.getFrom());
+                break;
+            case R.id.to:
+                try {
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java]: Having a `try`/`catch` block here makes the code harder to read and duplicates the error handling code. I suggest creating a new method that gets the recipients from a message and handles the exception.

File: k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java [hunk 4]
```diff
@@ -154,10 +185,31 @@ private void onAddSenderToContacts() {
         }
     }
 
+    private void onAddAddressesToClipboard(Address[] addresses){
+        if(mMessage != null){
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java]: This check is unnecessary because `mMessage` is not used inside the method. Also, by the time we get here `mMessage` will have been dereferenced in all code paths that lead here.

File: k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java [hunk 5]
```diff
@@ -154,10 +185,31 @@ private void onAddSenderToContacts() {
...
+            try{
+                String addressesToCopy = "";
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java]: You might want to use `Address.toString(Address[])` here. It returns a comma separated list of addresses.

If you want to continue using a space separated list please use a `StringBuilder` to construct that string.

File: k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java [hunk 6]
```diff
@@ -154,10 +185,31 @@ private void onAddSenderToContacts() {
...
+                for(Address addressTemp : addresses){
+                    addressesToCopy += addressTemp.getAddress() + " ";
+                }
+                addressesToCopy = addressesToCopy.substring(0,addressesToCopy.length()-1);
+                ClipboardManager.getInstance(mContext).setText("addresses", addressesToCopy);
+                Toast.makeText(mContext, R.string.copy_address_to_clipboard, Toast.LENGTH_LONG).show();
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java]: It would be nice to use quantity strings here. See http://developer.android.com/guide/topics/resources/string-resource.html#Plurals

File: k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java [hunk 7]
```diff
@@ -154,10 +185,31 @@ private void onAddSenderToContacts() {
...
+            }
+            catch (Exception e){
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java]: Catching all exceptions is a bad habit that quickly gets out of hand. K-9 Mail is especially guilty of that. Let's try not to make it worse.
We want unexpected exceptions to crash the app so we can learn about them and then deal with those exceptions explicitly.

**Comment by Riverside14** [comment]
That might be good now.

File: k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java [hunk 8]
```diff
@@ -154,11 +177,32 @@ private void onAddSenderToContacts() {
...
     }
 
+    public String createMessage(int addressesCount){
+        return mContext.getResources().getQuantityString(R.plurals.copy_address_to_clipboard,addressesCount);
+    }
+
+    private void onAddAddressesToClipboard(Address[] addresses){
+        StringBuilder addressesToCopy = new StringBuilder();
+        for(Address addressTemp : addresses){
+            addressesToCopy.append(addressTemp.getAddress() + " ");
```
Comment (philipwhiuk) [inline k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java]: I think it might be more useful to separate them with a `;` as that's the standard email separator.

Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java]: The separator in email headers is `,`.

File: k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java [hunk 9]
```diff
@@ -154,11 +177,32 @@ private void onAddSenderToContacts() {
         }
     }
 
+    public String createMessage(int addressesCount){
```
Comment (Valodim) [inline k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java]: code style

**Comment by cketti** [comment]
Squashed, cleaned up and merged: https://github.com/k9mail/k-9/compare/5fdc46386bb584dd74b4a479dadd46f34a234f2d...af7e0a1b70e178c6f11e21f3be721b67472de9b2
